### PR TITLE
Add Amplitude events to new Write OAuth flows C-2801

### PR DESCRIPTION
**Pull request by nicoback2**
### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/models/Analytics.ts [hunk 1]
```diff
@@ -84,6 +84,13 @@ export enum Name {
   AUDIUS_OAUTH_COMPLETE = 'Audius Oauth: Login (authenticate) Success',
   AUDIUS_OAUTH_ERROR = 'Audius Oauth: Login (authenticate) Failed',
 
+  // Developer app
+  DEVELOPER_APP_DELETE_ERROR = 'Developer Apps: Delete app error',
```
Comment (dylanjeffers) [inline packages/common/src/models/Analytics.ts:88]: nit but imo order create before delete, and then order each as submit success error

File: packages/common/src/models/Analytics.ts [hunk 2]
```diff
@@ -1417,25 +1424,63 @@ type AudiusOauthStart = {
...
 
 type AudiusOauthError = {
   eventName: Name.AUDIUS_OAUTH_ERROR
+  appId: string | string[]
+  scope: string | string[]
   isUserError: boolean
   error: string
 }
 
+type DeveloperAppDeleteSuccess = {
```
Comment (dylanjeffers) [inline packages/common/src/models/Analytics.ts:1454]: same here?

**Review by dylanjeffers — APPROVED** [review]
looks great from my end, but maybe get seb's review also :)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--oauth-tracking

**Pull request merged**